pkg/ebpf/bytecode/runtime/telemetry.go

```diff
@@ -0,0 +1,60 @@
+package runtime
+
+// Telemetry stores telemetry related to a particular type of runtime asset
+type Telemetry interface {
+	Get() map[string]int64
+}
+
+type noOpTelemetry struct{}
+func newNoOpTelemetry() Telemetry {
+	return &noOpTelemetry{}
+}
+func (t *noOpTelemetry) Get() map[string]int64 {
+	return map[string]int64{}
+}
+
+
+// CompilationResult enumerates runtime compilation success & failure modes
+type CompilationResult int
+const (
+	Success CompilationResult = iota
+	KernelVersionErr
+	VerificationError
+	OutputDirErr
+	OutputFileErr
+	NewCompilerErr
+	CompilationErr
+	ResultReadErr
+)
+
+type compilationTelemetry struct {
+	enabled bool
+	result   CompilationResult
+	duration int64
```

Comment:
Can we use a `time.Duration` here and cast only to `int64` in `Get()`?

Reply:
Definitely!